diff --git a/packages/opencode/src/plugin/index.ts b/packages/opencode/src/plugin/index.ts
index 1c129f60829..e69b87bc949 100644
--- a/packages/opencode/src/plugin/index.ts
+++ b/packages/opencode/src/plugin/index.ts
@@ -34,8 +34,8 @@ export namespace Plugin {
       project: Instance.project,
       worktree: Instance.worktree,
       directory: Instance.directory,
-      get serverUrl(): URL {
-        throw new Error("Server URL is no longer supported in plugins")
+      get serverUrl(): URL | undefined {
+        return undefined
       },
       $: Bun.$,
     }
diff --git a/packages/plugin/src/index.ts b/packages/plugin/src/index.ts
index b78bcae177d..f2a201fe7e1 100644
--- a/packages/plugin/src/index.ts
+++ b/packages/plugin/src/index.ts
@@ -28,7 +28,7 @@ export type PluginInput = {
   project: Project
   directory: string
   worktree: string
-  serverUrl: URL
+  serverUrl?: URL
   $: BunShell
 }
 
